File: src/main/java/phoenix/enity/TalpaEntity.java
package phoenix.enity;

import net.minecraft.entity.*;
import net.minecraft.entity.ai.goal.*;
import net.minecraft.entity.monster.AbstractRaiderEntity;
import net.minecraft.entity.passive.AnimalEntity;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.item.Items;
import net.minecraft.item.crafting.Ingredient;
import net.minecraft.util.DamageSource;
import net.minecraft.util.math.AxisAlignedBB;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.IWorld;
import net.minecraft.world.World;
import phoenix.init.PhoenixEntities;

public class TalpaEntity extends AnimalEntity
{
    private BlockPos boundOrigin;
    public TalpaEntity(EntityType<TalpaEntity> type, World worldIn)
    {
        super(type, worldIn);
    }

    public static TalpaEntity create(World worldIn)
    {
        return new TalpaEntity(PhoenixEntities.TALPA.get(), worldIn);
    }

    @Override
    public void tick()
    {
        this.setNoGravity(isEntityInsideOpaqueBlock());
        super.tick();
    }
    @Override
    protected void registerAttributes()
    {
        super.registerAttributes();
        this.getAttribute(SharedMonsterAttributes.MAX_HEALTH).setBaseValue(14.0D);
    }
    @Override
    public boolean isInvulnerableTo(   DamageSource source)
    {
        return super.isInvulnerableTo(source) || source == DamageSource.IN_WALL;
    }

    @Override
    public int getMaxFallHeight()
    {
        return 7;
    }

    public BlockPos getBoundOrigin() {
        return this.boundOrigin;
    }

    @Override
    public EntitySize getSize(   Pose poseIn)
    {
        return new EntitySize(0.6F, 0.6F, false);
    }

      
    @Override
    public AxisAlignedBB getCollisionBox(Entity entityIn)
    {
        return isEntityInsideOpaqueBlock() ? null : super.getCollisionBox(entityIn);
    }

    @Override
    public boolean canSpawn(IWorld worldIn, SpawnReason spawnReasonIn)
    {
        return this.getPosition().getY() > 10 && this.getPosition().getY() < 50 && super.canSpawn(worldIn, spawnReasonIn);
    }

    @Override
    protected void registerGoals()
    {
        this.goalSelector.addGoal(0, new SwimGoal(this));
        this.goalSelector.addGoal(1, new PanicGoal(this, 0.25D));
        this.goalSelector.addGoal(2, new BreedGoal(this, 0.5D));
        this.goalSelector.addGoal(3, new TemptGoal(this, 0.5D, Ingredient.fromItems(Items.WHEAT), false));
        this.goalSelector.addGoal(4, new FollowParentGoal(this, 0.5D));
        this.goalSelector.addGoal(9, new LookAtGoal(this, PlayerEntity.class, 3.0F, 1.0F));
        this.goalSelector.addGoal(10, new LookAtGoal(this, MobEntity.class, 8.0F));
        this.targetSelector.addGoal(5, (new HurtByTargetGoal(this, AbstractRaiderEntity.class)).setCallsForHelp());
        this.targetSelector.addGoal(6, new NearestAttackableTargetGoal<>(this, PlayerEntity.class, true));
    }
      
    @Override
    public AgeableEntity createChild(AgeableEntity ageable)
    {
        return TalpaEntity.create(ageable.world);
    }
}

File: src/main/java/phoenix/enity/TalpaEntity.kt
package phoenix.enity

import net.minecraft.entity.*
import net.minecraft.entity.ai.goal.*
import net.minecraft.entity.item.ItemEntity
import net.minecraft.entity.monster.AbstractRaiderEntity
import net.minecraft.entity.passive.AnimalEntity
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.item.Items
import net.minecraft.item.crafting.Ingredient
import net.minecraft.util.DamageSource
import net.minecraft.util.math.AxisAlignedBB
import net.minecraft.util.math.BlockPos
import net.minecraft.world.IWorld
import net.minecraft.world.World
import phoenix.init.PhoenixEntities


class TalpaEntity(type: EntityType<TalpaEntity>, worldIn: World) : AnimalEntity(type, worldIn)
{
    val boundOrigin: BlockPos? = null

    override fun tick()
    {
        setNoGravity(isEntityInsideOpaqueBlock)
        super.tick()
    }

    override fun registerAttributes()
    {
        super.registerAttributes()
        getAttribute(SharedMonsterAttributes.MAX_HEALTH).baseValue = 14.0
    }

    override fun isInvulnerableTo(source: DamageSource) = super.isInvulnerableTo(source) || source === DamageSource.IN_WALL
    override fun getMaxFallHeight() = 7
    override fun getSize(poseIn: Pose) = EntitySize(0.6f, 0.6f, false)
    override fun getCollisionBox(entityIn: Entity): AxisAlignedBB? = if (isEntityInsideOpaqueBlock) null else super.getCollisionBox(entityIn)
    override fun createChild(ageable: AgeableEntity): AgeableEntity = TalpaEntity(PhoenixEntities.TALPA.get(), ageable.world)
    override fun canSpawn(worldIn: IWorld, spawnReasonIn: SpawnReason) = this.position.y in 11..49 && super.canSpawn(worldIn, spawnReasonIn)

    override fun registerGoals()
    {
        goalSelector.addGoal(0, SwimGoal(this))
        goalSelector.addGoal(1, PanicGoal(this, 0.25))
        goalSelector.addGoal(2, BreedGoal(this, 0.5))
        goalSelector.addGoal(3, TemptGoal(this, 0.5, Ingredient.fromItems(Items.WHEAT), false))
        goalSelector.addGoal(4, FollowParentGoal(this, 0.5))
        goalSelector.addGoal(9, LookAtGoal(this, PlayerEntity::class.java, 3.0f, 1.0f))
        goalSelector.addGoal(10, LookAtGoal(this, MobEntity::class.java, 8.0f))
        targetSelector.addGoal(5, HurtByTargetGoal(this, AbstractRaiderEntity::class.java).setCallsForHelp())
        targetSelector.addGoal(6, NearestAttackableTargetGoal(this, PlayerEntity::class.java, true))
    }


}
